src/host/history.cpp

```diff
@@ -152,27 +153,28 @@ void CommandHistory::_Reset()
     return S_OK;
 }
 
-std::wstring_view CommandHistory::GetNth(const SHORT index) const
+std::wstring_view CommandHistory::GetNth(Index index) const
 {
-    try
+    if (index >= 0 && index < _commands.size())
     {
         return _commands.at(index);
     }
-    CATCH_LOG();
-
     return {};
 }
 
-[[nodiscard]] HRESULT CommandHistory::RetrieveNth(const SHORT index,
-                                                  std::span<wchar_t> buffer,
-                                                  size_t& commandSize)
+const std::vector<std::wstring>& CommandHistory::GetCommands() const noexcept
+{
+    return _commands;
+}
+
+[[nodiscard]] HRESULT CommandHistory::RetrieveNth(const Index index, std::span<wchar_t> buffer, size_t& commandSize)

```

Comment:
Ugh we have Get and Retrieve. Can we rename `Retrieve` to `Copy`? They're both synonyms otherwise, and I hate methods whose names are synonyms but behaviors are totally unrelated.

ALSO THIS UPDATES STATE WHERE GET DOESN'T.
AGH.

Reply:
If you don't mind, I'd rather not make such a change right now. It would make the diff quite a bit larger, which might not be a good timing, given that I'll very soon rewrite every other caller of these functions.

Reply:
Argh, I'll accept it!